File: Fall_For_One_Challenge/shadow_and_love.c
#include <stdio.h>
#include <stdlib.h>

int sum_div();
void swap(int *a, int *b);
int sum(int arr[]);

int ARR_SIZE;
int A, B, SA, SB, PS, SL;
int L, R;

int main()
{
    int t, res;
    scanf("%d", &t);

    while(t--)
    {
        scanf("%d %d %d %d", &A, &B, &L, &R);
        sum_div();
        PS = SA + SB;
        res = PS - SL;
        if(res < 0)
        {
            res *= -1;
        }

        if(res%2 == 0 && res%3 == 0)
        {
            printf("TRUE LOVE\n");
        }
        else if(PS%2 == 0 || PS%3 == 0)
        {
            printf("LOVE\n");
        }
        else
        {
            printf("FAKE LOVE\n");
        }
        
    }
}

int sum_div()
{
    int num;
    // LCM = lcm(A, B);

    if(R < L)
    {
        swap(&R, &L);
    }

    for(num = L; num <= R; num++)
    {
        if(num%A == 0)
        {
            SA += num;
            // printf("%d,: %d ", num, A);
        }
        if(num%B == 0)
        {
            SB += num;
            // printf("%d,: %d ", num, B);
        }
        if(num%A == 0 && num%B == 0)
        {
            SL += num;
            // printf("%d,: %d ", num, LCM);
        }
    }
    // ARR_SIZE = i;

    return 1;
    // return sum(factors);
}

void swap(int *a, int *b)
{
    int temp;
    temp = *a;
    *a = *b;
    *b = temp;

    return;
}

File: Fall_For_One_Challenge/shadow_and_love.py
#!/usr/bin/python3

def sum_div(n, L, R) : # Sum of numbers divisible by n
    if R <  L :
        R, L = L, R
    factors = list()
    for num in range(L, R+1) :
        if num % n == 0 :
            factors.append(num)
    return sum(factors)

def gcd(x, y):
   while y :
       x, y = y, x % y

   return x

def lcm(x, y):
   lcm = (x*y)//gcd(x,y)
   return lcm

try :
    test_cases = int(input())
except :
    quit()

for case in range(test_cases) :
    A, B, L, R = input().split()
    SA = sum_div(int(A), int(L), int(R))
    SB = sum_div(int(B), int(L), int(R))
    PS = SA + SB
    LCM = lcm(int(A), int(B))
    SL = sum_div(LCM, int(L), int(R))
    print(SA, SB, PS, LCM, SL, abs(PS-SL))
    if (abs(PS-SL) % 2 == 0) and (abs(PS-SL) %3 == 0) :
        print("TRUE LOVE")
    elif (PS % 2 == 0) or (PS % 3 == 0) :
        print("LOVE")
    else :
        print("FAKE LOVE")

# def lcm(x, y):
#    if x > y:
#        greater = x
#    else:
#        greater = y

#    while True:
#        if (greater % x == 0) and (greater % y == 0) :
#            lcm = greater
#            break
#        greater += 1

#    return lcm